trie/Trie.java

```diff
@@ -0,0 +1,115 @@
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Implement a trie with insert, search, and starts with methods using hash map.
+ */
+class TrieNode {
+    public char ch;
+    // has map store trie node
+    public HashMap<Character, TrieNode> children = new HashMap<Character, TrieNode>();
+    public boolean isLeaf;
+
+    public TrieNode() {
+    }
+
+    public TrieNode(char ch) {
+        this.ch = ch;
+    }
+}
+
+public class Trie {
+    private TrieNode root;
+    // constructor
+    public Trie() {
+        root = new TrieNode();
+    }
+
+    /* Insert into the Trie
+     * args :
+     * key : String to insert into Trie
+     * Time Complexity: O(len(word))
+     */
+    public void insert(String word) {
+        HashMap<Character, TrieNode> children = root.children;
+        // insert word char by char
+        for (int i = 0; i < word.length(); i++) {
+            char ch = word.charAt (i);
+            TrieNode root;
+            /* Check status of that child node
+             * If it is empty, them fill it
+             * If it is present, them use this as the next root
+             */
+            if (children.containsKey(ch)) {
+                root = children.get(ch);
+            } else {
+                root = new TrieNode(ch);
+                children.put(ch, root);
+            }
+            children = root.children;
+
+            //set leaf node
+            if (i == word.length() - 1)
+                root.isLeaf = true;
+        }
+    }
+
+    /* Searches into the Trie
+     * args:
+     * key : string to search into Trie
+     * Time complexity : O(len(word))
+     */
+    public boolean search(String word) {
+        TrieNode root = searchNode(word);
+        // if valid words return true
+        if (root != null && root.isLeaf) {
+            return true;
+        }
+        return false;
+    }
+
+    /** Returns if there is any word in the trie
+     * that start with the given prefix.
+     */
+    public boolean startsWith(String prefix) {
+        // words not start with prefix then return false
+        if (searchNode(prefix) == null) {
```

Comment:
This can be written as `return searchNode(prefix) != null)` thats why I guess the codacy is pointing error.